File: Electrical Outlets/CodePal/Python/Electrical.py
def read_input():
    """
    Reads the input data from standard input and returns it as a matrix of cord lengths.
    """
    n = int(input())
    cords = []
    for i in range(n):
        row = list(map(int, input().split()))
        cords.append(row)
    return cords

def create_music_stands():
    """
    Creates a list of all the music stands, with the wall outlet as stand 0.
    """
    return list(range(n))

def create_branch_sockets(music_stands):
    """
    Creates a list of all the possible branch sockets, which are the music stands that are not the wall outlet.
    """
    return music_stands[1:]

def create_cords(music_stands, cords):
    """
    Creates a list of all the possible cords, which are the connections between each pair of music stands.
    """
    all_cords = []
    for i in range(len(music_stands)):
        for j in range(i+1, len(music_stands)):
            cord = (cords[i][j], music_stands[i], music_stands[j])
            all_cords.append(cord)
    return sorted(all_cords)

def build_network(cords, branch_sockets):
    """
    Builds a network of cords with power outlets that use as little cord (power cable) as possible and no unused power outlets.
    """
    network = []
    while len(branch_sockets) > 0:
        cord = cords.pop(0)
        if cord[1] in branch_sockets and cord[2] not in branch_sockets:
            network.append(cord)
            branch_sockets.remove(cord[1])
    return network

def print_output(network):
    """
    Prints the network of cords to standard output in the specified format.
    """
    for cord in network:
        length = cord[0]
        sockets = 2
        start = cord[1]
        end = cord[2]
        print(f"{length} {sockets} {start} {end}")

# Main program
cords = read_input()
music_stands = create_music_stands()
branch_sockets = create_branch_sockets(music_stands)
all_cords = create_cords(music_stands, cords)
network = build_network(all_cords, branch_sockets)
print_output(network)

File: Electrical Outlets/GitHub-Copilot/Java/Electrical.java
import java.util.Arrays;
import java.util.Scanner;

public class Electrical {
    //In a concert hall, the musicians sit in designated places, and each music stand has its own lamp with a lamp cord that just barely reaches the floor. In order for all music stand lamps to be powered, an electrician needs to build a network of branch sockets and cords, where the number of sockets and the length of the cords are adjusted by the electrician according to need. There needs to be an electrical outlet under each music stand, and branch sockets are only allowed to be placed under the music stands. There is only one wall outlet in the concert hall, and it sits 150 centimeters from the nearest music stand. The problem is to design a network of cords with power outlets that use as little cord (power cable) as possible and no unused power outlets. 
    //The input data is given through standard input on the following format. The first row contains N, the number of music stands. The following rows are each a row of the NxN matrix, with the numbers separated by spaces, and place (i, j) indicates how long a cord would be required to be to reach from stand i to stand j. The Nth music stand is the one closest to the wall outlet. The length of the cords are positive integers, representing the length in centimeters. 
    //The output data should be printed to standard output in the following format. A row for each of the branch sockets used to build the network that powers the lamps. Each row should print the length of the cord, the number of sockets it has, the number of the music stand where the cord is plugged in, and the number of the music stand it lies beneath, all separated by spaces. The wall outlet is considered as music stand 0.
    public static void main(String[] args){

        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[][] cords = new int[n][n];
        for(int i = 0; i < n; i++){
            for(int j = 0; j < n; j++){
                cords[i][j] = sc.nextInt();
            }
        }
        sc.close();
        int[] distances = new int[n];
        boolean[] used = new boolean[n];
        Arrays.fill(distances, Integer.MAX_VALUE);
        distances[0] = 0;
        int[] parent = new int[n];
        Arrays.fill(parent, -1);
        for(int i = 0; i < n - 1; i++){
            int u = -1;
            for(int j = 0; j < n; j++){
                if(!used[j] && (u == -1 || distances[j] < distances[u])){
                    u = j;
                }
            }
            used[u] = true;
            for(int v = 0; v < n; v++){
                if(!used[v] && cords[u][v] < distances[v]){
                    distances[v] = cords[u][v];
                    parent[v] = u;
                }
            }
        }
        for(int i = 1; i < n; i++){
            System.out.println(distances[i] + " 2 " + parent[i] + " " + i);
        }

    }
}